airflow/contrib/operators/dataflow_operator.py

```diff
@@ -128,6 +128,110 @@ def execute(self, context):
         hook.start_java_dataflow(self.task_id, dataflow_options, self.jar)
 
 
+class DataflowTemplateOperator(BaseOperator):
+    """
+    Start a Templated Cloud DataFlow batch job. The parameters of the operation
+    will be passed to the job.
+    It's a good practice to define dataflow_* parameters in the default_args of the dag
+    like the project, zone and staging location.
+    https://cloud.google.com/dataflow/docs/reference/rest/v1b3/LaunchTemplateParameters
+    https://cloud.google.com/dataflow/docs/reference/rest/v1b3/RuntimeEnvironment
+    ```
+    default_args = {
+        'dataflow_default_options': {
+            'project': 'my-gcp-project'
+            }
+        'environment': {
```

Comment:
Does it make sense to merge environment into dataflow_default_options or options, to be consistent with other Dataflow*Operator? 

You could reconstruct environment in the hook.

Reply:
I agree with @fenglu-g on this one. Merging the environment stuff into some dataflow specific root key is preferable, since the values in the default args variable are used across all operators in a DAG (including non-Dataflow ones). Having a generic "environment" variable in there is a bit confusing, to say the least.

Reply:
Thanks, I originally started that way but thought it may be better to keep the sections separated, I can see it will actually be confusing.
Will reconstruct accordingly.

Reply:
R: @fenglu-g @criccomini 
Thanks for the feedback, added the below (Let me know your views please):
- Validate projectId is set, raise exception otherwise
- Jobname: name change applied
- Environment variables merged into dataflow_default_options
- Tests consolidated